File: demo/MoffatBackend/src/main/java/com/csd460/moffat/model/reservation/Reservation.java
package com.csd460.moffat.model.reservation;

import com.csd460.moffat.model.room.Room;
import com.csd460.moffat.model.user.User;

import java.sql.Timestamp;


public class Reservation {
    public long reservationId;
    public User user;
    public Room room;
    public Timestamp checkInDate;
    public Timestamp checkOutDate;

    public Reservation(long reservationId, User user, Room room, Timestamp checkInDate, Timestamp checkOutDate) {
        this.reservationId = reservationId;
        this.user = user;
        this.room = room;
        this.checkInDate = checkInDate;
        this.checkOutDate = checkOutDate;
    }

    public Reservation(User user, Room room, Timestamp checkInDate, Timestamp checkOutDate) {
        this.user = user;
        this.room = room;
        this.checkInDate = checkInDate;
        this.checkOutDate = checkOutDate;
    }

    public long getReservationId() {
        return reservationId;
    }

    public void setReservationId(long reservationId) {
        this.reservationId = reservationId;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public Timestamp getCheckInDate() {
        return checkInDate;
    }

    public void setCheckInDate(Timestamp checkInDate) {
        this.checkInDate = checkInDate;
    }

    public Timestamp getCheckOutDate() {
        return checkOutDate;
    }

    public void setCheckOutDate(Timestamp checkOutDate) {
        this.checkOutDate = checkOutDate;
    }

    @Override
    public String toString() {
        return "Reservation{" +
                "reservationId=" + reservationId +
                ", user=" + user +
                ", room=" + room +
                ", checkInDate=" + checkInDate +
                ", checkOutDate=" + checkOutDate +
                '}';
    }
}

File: src/frontend/public/model/reservation.php
<?php

require_once 'room.php';
require_once 'user.php';

class Reservation {
    public int $reservation_id;
    public User $user;
    public Room $room;
    public $checkin_date;
    public $checkout_date;

    /**
     * @param int $reservation_id
     * @param User $user
     * @param Room $room
     * @param $checkin_date
     * @param $checkout_date
     */
    public function __construct(int $reservation_id, User $user, Room $room, $checkin_date, $checkout_date)
    {
        $this->reservation_id = $reservation_id;
        $this->user = $user;
        $this->room = $room;
        $this->checkin_date = $checkin_date;
        $this->checkout_date = $checkout_date;
    }

    /**
     * @return int
     */
    public function getReservationId(): int
    {
        return $this->reservation_id;
    }

    /**
     * @param int $reservation_id
     */
    public function setReservationId(int $reservation_id): void
    {
        $this->reservation_id = $reservation_id;
    }

    /**
     * @return User
     */
    public function getUser(): User
    {
        return $this->user;
    }

    /**
     * @param User $user
     */
    public function setUser(User $user): void
    {
        $this->user = $user;
    }

    /**
     * @return Room
     */
    public function getRoom(): Room
    {
        return $this->room;
    }

    /**
     * @param Room $room
     */
    public function setRoom(Room $room): void
    {
        $this->room = $room;
    }

    /**
     * @return mixed
     */
    public function getCheckinDate()
    {
        return $this->checkin_date;
    }

    /**
     * @param mixed $checkin_date
     */
    public function setCheckinDate($checkin_date): void
    {
        $this->checkin_date = $checkin_date;
    }

    /**
     * @return mixed
     */
    public function getCheckoutDate()
    {
        return $this->checkout_date;
    }

    /**
     * @param mixed $checkout_date
     */
    public function setCheckoutDate($checkout_date): void
    {
        $this->checkout_date = $checkout_date;
    }




}